packages/victory-core/src/victory-util/data.js

```diff
@@ -91,6 +92,64 @@ function cleanData(dataset, props) {
   });
 }
 
+// This method will remove data points that fall outside of the desired domain (non-continuous charts only)
+function formatDataFromDomain(dataset, props) {
+  // domain for checking data points that are outside of the bounds
+  // symbol to determine chart types with a single y value point
+  // interpolation for determining chart types with continuous data points that shouldn't be altered
+  // errorX/Y for determining chart types with error whiskers that shouldn't be altered here
+  const { domain, symbol, interpolation, errorX, errorY } = props;
+
+  if (interpolation || errorX || errorY) return dataset;
+
+  const minDomainX =
+    !domain || !domain.x ? Domain.getMinFromProps(props, "x") : Collection.getMinValue(domain.x);
+  const maxDomainX =
+    !domain || !domain.x ? Domain.getMaxFromProps(props, "x") : Collection.getMaxValue(domain.x);
+  const minDomainY =
+    !domain || !domain.y ? Domain.getMinFromProps(props, "y") : Collection.getMinValue(domain.y);
+  const maxDomainY =
+    !domain || !domain.y ? Domain.getMaxFromProps(props, "y") : Collection.getMaxValue(domain.y);
+
+  const exists = (val) => val !== undefined;
+
+  // eslint-disable-next-line complexity
+  return dataset.map((datum) => {
+    let { _x, _y, _y0, _y1 } = datum;
+
+    // don't alter data points if _y is an array of values
+    if (Array.isArray(_y)) return datum;
+
+    // single x point less than min domain
+    if (exists(_x) && _x < minDomainX) _x = null;
+    // single x point greater than max domain
+    if (exists(_x) && _x > maxDomainX) _x = null;
+
+    // single y point less than min domain
+    if (exists(_y) && !exists(_y0) && _y < minDomainY) {
+      if (exists(symbol)) _y = null;
```

Comment:
I don't like duck-typing on the `symbol` prop here to check for `VictoryScatter`. I think `_y` can be handled the same for all chart types after checking on the existence of `_y0` and `_y1`